### PR TITLE
Added configurable bastion types

### PR DESCRIPTION
**Pull request by TealNerd**
Also switched to civmodcore because why not? duplicate code is dumb

added better humbug check to check once rather than every time

### CODE REVIEW

File: Bastion/src/isaac/bastion/manager/BastionBlockManager.java [hunk 1]
```diff
@@ -77,27 +73,9 @@ public void erodeFromTeleport(Location loc, String player, Set<BastionBlock> blo
...
-				BastionBlock toErode = ordered.get(erode);
-				if (fromBlock) {
-					toErode.erode(toErode.erosionFromBlock());
-				} else {
-					toErode.erode(toErode.erosionFromPearl());
-				}
-				ordered.remove(erode);
-			}
+		for(BastionBlock bastion : blocking) {
+			double erosion = fromBlock ? bastion.erosionFromBlock() : bastion.erosionFromPearl();
```
Comment (ProgrammerDan) [inline Bastion/src/isaac/bastion/manager/BastionBlockManager.java]: No. The prior code was implemented correctly, although inefficient; this change represents a significant regression (loss of functionality).

File: Bastion/src/isaac/bastion/manager/BastionBlockManager.java [hunk 2]
```diff
@@ -399,33 +374,6 @@ public void handleEnderPearlLanded(PlayerTeleportEvent event) {
 			event.setCancelled(true);
 			return;
 		}
-
```
Comment (ProgrammerDan) [inline Bastion/src/isaac/bastion/manager/BastionBlockManager.java]: Maybe I'm missing something but removing this entire section was not a good idea. What's the rationale?

Comment (ProgrammerDan) [inline Bastion/src/isaac/bastion/manager/BastionBlockManager.java]: To expand; the prior section blocks pearls based on _where they land_ while the second section you removed blocks pearls based on _where they were launched from_ ... e.g. cancels a pearl event if it originates in a hostile bastion field. Both are necessary. They are not identical.

**Comment by ProgrammerDan** [comment]
Just a few regressions looks like, otherwise seems ok, assuming the QTBox in CivModCore is equivalent.

File: Bastion/src/isaac/bastion/manager/BastionBlockManager.java (hunk 2 above)
Comment (TealNerd) [inline Bastion/src/isaac/bastion/manager/BastionBlockManager.java]: ah ok didn't realize that

**Pull request closed without merging**